src/itype.h

```diff
@@ -43,6 +43,20 @@ std::string ammo_name(std::string const &t);
 // Returns the default ammo for a category of ammo (e.g. ""00_shot"")
 std::string const& default_ammo(std::string const &guntype);
 
+struct islot_tool {
+    std::string ammo_id = "NULL";
+
+    itype_id revert_to = "null";
+    std::string revert_msg;
+
+    std::string subtype;
+
+    long max_charges = 0;
+    long def_charges = 0;
+    unsigned char charges_per_use = 0;
```

Comment:
Any reason this cannot be come `long` or `int`?

Incidentally at what point can we consider migrating `charges` from `long` to `int`?